src/Sulu/Component/Content/Document/Subscriber/OrderSubscriber.php

```diff
@@ -64,19 +73,18 @@ public function handlePersist(PersistEvent $event)
 
         $propertyName = $this->encoder->systemName(self::FIELD);
 
-        if ($node->hasProperty($propertyName)) {
-            return;
-        }
-
         $parent = $node->getParent();
-        $nodeCount = count($parent->getNodes());
-        $order = ($nodeCount + 1) * 10;
+        $count = 0;
+        foreach ($parent->getNodes() as $childNode) {
```

Comment:
Because it *is* more efficient in other cases where we would need to
call the registry multiple times from different subscribers in
more common, performance sensative, operations. I believe it is
important also for other reasons, but can't remember now.

On Thu, Jul 16, 2015 at 07:24:26AM -0700, Daniel Rotter wrote:
>    In [1]src/Sulu/Component/Content/Document/Subscriber/OrderSubscriber.php:
> 
>  >          $parent = $node->getParent();
>  > -        $nodeCount = count($parent->getNodes());
>  > -        $order = ($nodeCount + 1) * 10;
>  > +        $count = 0;
>  > +        foreach ($parent->getNodes() as $childNode) {
> 
>    [2]@dantleech Why don't we always do it like this then? IMO it would be
>    better to remove this handleNodeFromRegistry method then, and only get the
>    node if the given subscriber really needs it.
> 
>    —
>    Reply to this email directly or [3]view it on GitHub.
> 
>    Reverse link: [4]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu-io/sulu/pull/1378#discussion_r34792290
>    2. https://github.com/dantleech
>    3. https://github.com/sulu-io/sulu/pull/1378/files#r34792290
>    4. https://github.com/sulu-io/sulu/pull/1378/files#r34792290